merkle/compact_merkle_tree_test.go

```diff
@@ -17,6 +18,40 @@ func getTree() *CompactMerkleTree {
 	return NewCompactMerkleTree(NewRFC6962TreeHasher(trillian.NewSHA256()))
 }
 
+func (c *CompactMerkleTree) checkUnusedNodesInvariant() error {
```

Comment:
Since the test is in the same package you should have access to the non-exported contents of structs, so this func could just take a `*CompactMerkleTree` and reach in, rather than adding this testing method to CMT.